File: src/main/java/xyz/aeolia/ashutils/task/VanishTask.java
package xyz.aeolia.ashutils.task;

import net.ess3.api.events.VanishStatusChangeEvent;
import org.bukkit.scheduler.BukkitRunnable;
import xyz.aeolia.ashutils.user.UserHelper;

public class VanishTask extends BukkitRunnable {
  VanishStatusChangeEvent event;
  public VanishTask(VanishStatusChangeEvent e) {
    event = e;
  }

  @Override
  public void run() {
    UserHelper.getUser(event.getAffected().getUUID()).setVanish(event.getValue());
  }
}

File: src/main/java/xyz/aeolia/ashutils/task/VanishTask.kt
package xyz.aeolia.ashutils.task

import net.ess3.api.events.VanishStatusChangeEvent
import org.bukkit.scheduler.BukkitRunnable
import xyz.aeolia.ashutils.user.UserHelper

class VanishTask(val event: VanishStatusChangeEvent) : BukkitRunnable() {
  override fun run() {
    UserHelper.getUser(event.affected.uuid).vanish = event.value;
  }
}